test/btfgen.bats

```diff
@@ -0,0 +1,22 @@
+setup() {
+    load 'test_helper/bats-support/load'
+    load 'test_helper/bats-assert/load'
+    load 'test_helper/bats-file/load'
+
+    DIR="$( cd "$( dirname "$BATS_TEST_FILENAME" )" >/dev/null 2>&1 && pwd )"
+    PATH="$DIR/../tools:$PATH"
+}
+
+@test "run btfgen.sh with example bpf object for x86_64" {
+    run btfgen.sh -a x86_64 -o test/goldens/example.bpf.o
+    assert_success
+    diff custom-archive/ubuntu/20.04/x86_64/5.8.0-63-generic.btf test/goldens/custom-archive/x86_64/5.8.0-63-generic.btf
+    diff custom-archive/fedora/34/x86_64/5.11.12-300.fc34.x86_64.btf test/goldens/custom-archive/x86_64/5.11.12-300.fc34.x86_64.btf
```

Comment:
Note: I've included these two files for the timebeing but there's not any particular reason as to why I've chosen them.

Reply:
Hum.. I think we should have a gold image to be compared in the following cases:

- ubuntu 5.4, 5.8, 5.11 and 5.13 kernels
- fedora 5.3, 5.6, 5.8 and 5.11 kernels
- centos 8 (5.18.0-147.8, 4.18.0-193.28.1, 4.18.0-348.2.1)

This would cover cases for different types contained in those distribution kernels and their different versions. WDYT ?

Reply:
Added a bunch of more test cases.